kodi_addon_checker/pylint_checks.py

```diff
@@ -0,0 +1,31 @@
+import io
+import json
+import os
+
+from pylint.lint import Run
+from terminaltables import AsciiTable
+from pylint.reporters.json import JSONReporter
+
+from .report import Report
+from .record import Record, WARNING
+
+# TODO: Do not repeat name of same file in report just do something like . utils.py - 90,1780 ?
+table_format = [["File path", "Line no", "Message"]]
+
+
+def short_path(path):
+    return os.path.split(path)[1]
+
+
+def analyze(report: Report, filename: str):
+
+    ARGS = ["-r", "n", "--score=yes", "--lint-all=y", "--rcfile=kodi_addon_checker/.pylintrc"]
+    out = io.StringIO()
+    Run([filename] + ARGS, reporter=JSONReporter(out), do_exit=False)
+
+    json_data = json.loads(out.getvalue())
+    for issue in json_data:
+        table_format.append([short_path(issue['path']), issue['line'], issue['message']])
+
+    table = AsciiTable(table_format).table
+    report.add(Record(WARNING, table))
```

Comment:
I don't think this is the correct approach. The linter will report issues with different degrees of severity so the tool cannot simply return a warning with all the issues in a table. Ideally the tool should either calculate a "quality gate" or at least changing the record level accordingly. For example, in pylint, `Exxxx` messages mean `(syntax error raised for a module) and `Fxxxx` messages mean `(error prevented analysis)`. If pylint finds any of those messages, the reporter should log it as `PROBLEM` instead of warning

Reply:
we are still to decide what type of errors need to be caught. 
In this we are doing checks for all the pylint codes. But once we are final about the codes then I'll make the changes accordingly